README.md

```diff
@@ -496,6 +496,204 @@ We expect some apps will need some features that are not currently available:
+
+You will also likely want to have a utility to mock the `AuthUser` object that is passed around via the hooks and HOFs of `next-firebase-auth`. Given that this is not a direct mock of the library itself, it might make sense to put inside a `utils` folder or anywhere else that makes sense for your team.
+
+```javascript
+// Create a mock FirebaseUser instance with the fields that you use.
+const mockFirebaseUser = {
+	displayName: 'Banana Manana',
+  // ... other fields from firebaseUser that you may use
+};
+
+/**
+ * Build and return a dummy AuthUser instance to use in tests.
+ *
+ * @arg {boolean} isLoggedIn - Pass `false` to mimic a logged out user.
+ * @returns {AuthUserContext} - A mocked AuthUser instance, with 'serialize' added.
+ */
+const getMockAuthUser = (isLoggedIn = true) => ({
+  id: isLoggedIn ? 'abcd1234' : null,
+  email: isLoggedIn ? 'banana@banana.com' : null,
+  emailVerified: isLoggedIn,
+  getIdToken: jest.fn(async () => (isLoggedIn ? 'i_am_a_token' : null)),
+  clientInitialized: isLoggedIn,
+  firebaseUser: isLoggedIn ? mockFirebaseUser : null,
+  signOut: jest.fn(),
+  serialize: jest.fn(() => 'serialized_auth_user'),
+});
+
+export default getMockAuthUser;
+```
+
+Once the mocks are included, then each test suite will need to import the functions that the component calls. Additionally, the component that is being tested needs to be `require`d inside a `beforeEach` function or inside each test case -- this is because the mocking of all the auth methods for `next-firebase-auth` has to happen _before_ your component is imported, because the call to the `next-firebase-auth` function is part of the default export of your component (eg. `export default withAuthUser()(MyComponent)`). Here is a full-fledged example of a working test suite.
+
+Given the following component:
+
+```javascript
+import React from 'react';
+import { useAuthUser, withAuthUser } from 'next-firebase-auth';
+
+function UserDisplayName() {
+  const AuthUser = useAuthUser();
+  const { displayName = 'anonymous' } = AuthUser.firebaseUser;
+  return (
+    <span>{displayName}</span>
+  );
+}
+
+export default withAuthUser()(UserDisplayName);
+```
+
+you can write a test suite like this:
+
+```javascript
+import { render, screen } from '@testing-library/react';
+
+// Import the functions that the component module calls, which allows jest to mock them
+// in the context of this test run. This allows you to manipulate the return value of each
+// function within this test suite.
+import { useAuthUser, withAuthUser } from 'next-firebase-auth';
+
+// Import your mock AuthUser generator
+import getMockAuthUser from '../../utils/test-utils/get-mock-auth-user';
+
+// Tell jest to mock the whole library, which will basically replace it with the default export
+// from `__mocks__/next-firebase-auth/`.
+jest.mock('next-firebase-auth');
```

Comment:
If using [manual mocks for Node modules](https://jestjs.io/docs/manual-mocks#mocking-node-modules), they'll be automatically mocked.